### DISCUSSION TITLE
Can I clone a private repo with private key of the GitHub App which is installed for the repo?

### DISCUSSION THREAD

**Discussion by dongho-jung** [community/community#28291] [Other Feature Feedback, Questions, & Ideas] [answered]
What I want to do was cloning a private repo and I've been using deploy key. However, deploy key is somewhat cumbersome because I need to make ssh key pair and add deploy key and secret every time. Instead, I'm trying another approach, create a Github App and install it to repos I want. then, from other servers, clone the repos with private key of the app. But I'm not sure it's possible or not...

I know I might be able to do something with REST API and JWT that can be made from the private key([example](https://gist.github.com/pelson/47c0c89a3522ed8da5cc305afc2562b0)) though, I want to clone directly with the private key without involving any programming languages like `GIT_SSH_COMMAND='ssh -i ./foobar-deploy-app.2022-08-03.private-key.pem -o IdentitiesOnly=yes' git clone git@github.com:0xf4d3c0d3/foo.git`. Of course, foobar-deploy-app.2022-08-03.private-key.pem is the private key of the GitHub App : )

---
In summary,

1. There are private repos foo and bar
2. There is a GitHub App installed at those repos
3. The GitHub App has private key and permissions for reading those repos' content
4. Clone those repos with (3)'s private key, like `GIT_SSH_COMMAND='ssh -i ./foobar-deploy-app.2022-08-03.private-key.pem -o IdentitiesOnly=yes' git clone git@github.com:0xf4d3c0d3/foo.git` <- is that any possible?

**Comment by gm3dmo** [comment, marked as answer]
The private key for a GitHub App cannot be used in the way you have described.

**Reply by dongho-jung** [reply]
Thanks for your answer

**Reply by souenzzo** [reply]
Why in "deploy keys" section it says 

> We recommend using GitHub Apps instead for fine grained control over repositories and enhanced security.

it gives the impression that github apps can replace deploy keys.
